holo-bgp/src/packet/message.rs

```diff
@@ -1001,11 +1013,14 @@ pub(crate) fn encode_ipv6_prefix(buf: &mut BytesMut, prefix: &Ipv6Network) {
     buf.put(&prefix_bytes[0..plen_wire]);
 }
 
-pub(crate) fn decode_ipv4_prefix(
+pub fn decode_ipv4_prefix(
     buf: &mut Bytes,
 ) -> DecodeResult<Option<Ipv4Network>> {
     // Parse prefix length.
-    let plen = buf.get_u8();
+    let plen = match buf.try_get_u8() {
```

Comment:
Couldn't we use `let plen = buf.try_get_u8().map_err(UpdateMessageError::InvalidNetworkField);`?

Reply:
Done.